src/main/clojure/clara/rules/logger.cljc

```diff
@@ -0,0 +1,25 @@
+(ns clara.rules.logger)
```

Comment:
A binding in an accumulator constraint will cause accumulation to occur over each value of that binding, so even if it is a new binding that isn't used elsewhere it actually does impact rule semantics.  So these two rules don't behave the same way:

```
(defrule get-current-temperature
  [?current-temp <- newest-temp :from [Temperature (= ?location location)]]
  =>
  (insert! (->CurrentTemperature ?current-temp))
  )
```

```
(defrule get-current-temperature
  [?current-temp <- newest-temp :from [Temperature]]
  =>
  (insert! (->CurrentTemperature ?current-temp))
  )
```

http://www.clara-rules.org/docs/accumulators/ discusses this, specifically the paragraph after the first example.   So I think we need logic that excludes from this check bindings that only appear once, but where that appearance is in an accumulator.

Reply:
Ah! I'm glad you pointed this out. I should have paid more attention. Can't this be the case for more than just an accumulator then? This may make the whole effort here need to shift to something within the compiler more than an external pass over the production structures as currently being done. 

I thought this external "linting" approach was fine since we could just ignore nearly all structure and find variables. However, seeing this accumulator case reminds me there are more semantics to perhaps be aware of. The detection may be more "context dependent" than I thought...

Reply:
Interesting corner case, I will write some code to handle the accumulator approach. Maybe if something else appear that is more dependent of context, we can revisit this approach of external linting.